site/content/integrate/apps/structure/bindings/embedded/_index.md

```diff
@@ -1,29 +1,190 @@
 
+## Post a message using HTTP REST
+
+To post a message using HTTP REST, the {{<newtabref title="Mattermost REST API" href="https://api.mattermost.com">}} is used.
+The {{<newtabref title="API endpoint" href="https://api.mattermost.com/#tag/posts/operation/CreatePost">}} responsible for posting a message is `/api/v4/posts`.
+
+Requests to the REST API must include an `Authorization` header with a bearer token set to the acting user or bot token.
+
+## Embedded bindings using HTTP REST
+
+The following example shows an HTTP REST request that posts a message with embedded bindings:
+
+{{<collapse id="embedded-rest" title="Example HTTP request">}}
+```http request
+POST /api/v4/posts HTTP/1.1
+Host: mattermost:8066
+Authorization: Bearer <acting_user_or_bot_token>
+Content-Type: application/json
+
+{
+    "channel_id": "<channel_id>",
+    "props": {
+        "app_bindings": [
+            {
+                "app_id": "<app_id>",
+                "location": "embedded",
+                "description": "Select your favourite coffee roast",
+                "bindings": [
+                    {
+                        "location": "coffee-roast",
+                        "label": "Coffee roast",
+                        "bindings": [
+                            {
+                                "location": "dark-roast",
+                                "label": "Dark roast",
+                                "submit": {
+                                    "path": "/set-roast-preference"
+                                }
+                            },
+                            {
+                                "location": "medium-roast",
+                                "label": "Medium roast",
+                                "submit": {
+                                    "path": "/set-roast-preference"
+                                }
+                            },
+                            {
+                                "location": "light-roast",
+                                "label": "Light roast",
+                                "submit": {
+                                    "path": "/set-roast-preference"
+                                }
+                            }
+                        ]
+                    },
+                    {
+                        "location": "i-dont-like-coffee",
+                        "label": "I don't like coffee",
+                        "submit": {
+                            "path": "/set-roast-preference"
+                        }
+                    }
+                ]
+            }
+        ]
+    }
+}
+```
+{{</collapse>}}
+
+## Embedded bindings using Golang
+
+{{<collapse id="golang-rest-example" title="Golang HTTP REST example">}}
+```go
+// Construct the post message with embedded bindings using Golang maps and slices
+post := map[string]any{
```

Comment:
I think the HTTP example above is enough, and this non-driver Golang example may be overkill

Reply:
I'd agree with @mickmister - we probably only need one http example and it should be the non-golang one.  A Golang developer would probably use the driver which is documented below.

There is a screenshot of the golang driver example.  Do we have an equivalent screenshot for the coffee roast example?

Reply:
Yes. I've updated the screenshots and examples, and I have removed the Golang HTTP example.

Reply:
Looks good.